mesonpy/__init__.py

```diff
@@ -733,11 +779,29 @@ def _wheel_builder(self) -> _WheelBuilder:
             self._copy_files,
         )
 
+    @property
+    def source_dir(self) -> pathlib.Path:
+        return self._source_dir
+
+    @property
+    def working_dir(self) -> pathlib.Path:
+        return self._working_dir
+
+    @property
+    def build_dir(self) -> pathlib.Path:
+        return self._build_dir
```

Comment:
Why expose these as properties? I don't see them used outside private code, and there accessing the property directly is fine. I'm not a big fan of indirection if not needed.